futurex_openedx_extensions/helpers/custom_roles.py

```diff
@@ -0,0 +1,75 @@
+"""New roles for FutureX Open edX Extensions."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from common.djangoapps.student.admin import CourseAccessRoleForm
+from common.djangoapps.student.roles import REGISTERED_ACCESS_ROLES, CourseRole, OrgRole, RoleBase
+
+from futurex_openedx_extensions.helpers.exceptions import FXCodedException, FXExceptionCodes
+
+log = logging.getLogger(__name__)
+
+
+def register_custom_access_role(cls: Any) -> Any:
+    """
+    Decorator that adds the new access role to the list of registered access roles to be accessible in the Django admin.
+
+    Note: roles inheritances is not supported
+
+    :param cls: The class to register
+    :type cls: Any
+    """
+    def _hacky_update_django_admin_choices(choices: list[tuple[str, str]]) -> None:
```

Comment:
I had to do it like this because `CourseAccessRoleForm` class of `edx-platform` is always declared before our extension